### ISSUE TITLE
Memory issue with long running process

### ISSUE THREAD

**Issue by mariusz360** [MayorMonty/Snoostorm#35]
**Describe the bug**
I'm running a continuous snoostorm process to monitor reddit comments and am running into memory issues. It looks like every new comment encountered is pushed into an array that isn't ever reduced in size. Perhaps snoostorm wasn't intended for a continuously running use-case?

**To Reproduce**
1. Execute a script that streams comments
2. Monitor the memory on your machine (I'm using `free -m` on Ubuntu)
3. Free memory should continue to be depleted

**Expected behavior**
Clear old inventory when it's no longer needed. The inventory array is used to filter out comments we have already seen. We can probably clear old inventory after the array reaches some length or remove inventory that's older than the 100th oldest comment.

Here's what I'm doing as a workaround. Probably not ideal:
```const client = new snoowrap(credentials)
let stream = new snoostorm.CommentStream(client, {
  subreddit: "all",
  limit: 100,
  pollTime: 1000,
}).on("item", (comment) => {
  processComment(comment)

  // Picked the #s 1000 and 500 here without much thought. Not ideal but better than
  // memory issues.
  let len = stream.inventory.length
  if (len > 1000) {
    stream.inventory = stream.inventory.slice(len - 500)
  }
})
```

**System Configuration:**
Using TypeScript? No
Node Version? v14.3.0
Snoostorm Version? v1.3.0

**Comment by MayorMonty** [comment]
The issue has been noted. My use case for snoostorm was designed for short-running processes (checking for modmail during a period), so this hasn't come up as much. I'm going through the process of rewriting all of my open source projects, so this design flaw will be fixed in `snoostorm@2.0.0`

**Comment by MayorMonty** [comment]
So in `snoostorm@1.4.0`, the generic `Poll` has been updated to only store an identifier of the specified object in a `Set`. 

```TypeScript
processed: Set<T[keyof T]> = new Set<T[keyof T]>();
```

If this still proves to be too much memory consumption, you can manually clear the processed list with `stream.processed.clear()`
